internal/http/services/owncloud/ocdav/locks.go

```diff
@@ -639,16 +643,19 @@ func (s *svc) unlockReference(ctx context.Context, _ http.ResponseWriter, r *htt
 		t = t[1 : len(t)-1]
 	}
 
-	switch err := s.LockSystem.Unlock(ctx, time.Now(), ref, t); err {
-	case nil:
-		return http.StatusNoContent, err
-	case errors.ErrForbidden:
-		return http.StatusForbidden, err
-	case errors.ErrLocked:
-		return http.StatusLocked, err
-	case errors.ErrNoSuchLock:
-		return http.StatusConflict, err
-	default:
-		return http.StatusInternalServerError, err
+	err := s.LockSystem.Unlock(ctx, time.Now(), ref, t)
+	if err == nil {
+		return http.StatusNoContent, nil
+	} else {
+		switch err.(type) {
+		case errtypes.Aborted, errtypes.Locked:
+			return http.StatusLocked, err
+		case errtypes.PermissionDenied:
+			return http.StatusForbidden, err
+		case errtypes.PreconditionFailed:
+			return http.StatusConflict, err
+		default:
+			return http.StatusInternalServerError, err
+		}
```

Comment:
why hide the switch in an if statement?
```go
switch err := s.LockSystem.Unlock(ctx, time.Now(), ref, t); err {
	case nil:
		return http.StatusNoContent, nil
	case errors.Aborted, errors.ErrLocked:
		return http.StatusLocked, err
	case errors.ErrForbidden:
		return http.StatusForbidden, err
	case errors.ErrNoSuchLock:
		return http.StatusConflict, err
	default:
		return http.StatusInternalServerError, err
}
```

Also, we need to make sure the LockSystem call only returns errors from the errors package, as that is what the interface description mentions. Can you ensure the implementation converts the errtypes errors to errors package errors? It is a necessary conversion.

Reply:
Added the Errtype mapping but kept `if err == nil` because it could be various codes 204, 200, etc.